PathSum2.java

```diff
@@ -0,0 +1,80 @@
+// Time Complexity : O(n)
```

Comment:
For this question, I was talking about the O(n^2) time complexity. new ArrayList<>(currPath) I believe is O(n) operation.